### PR TITLE
Feat/Add task

### PR DESCRIPTION
**Pull request by TIEN1909**
#175 Do adding task feature

**Result**

> Responsive screens

- X-Small devices (portrait phones, from 375px)

- Initial state
image

- After handle
image

- Small devices (landscape phones, 576px and up)

- Initial state
image

- After handle
image

### CODE REVIEW

File: javascript-training/big-practice/pomodoro-app/src/scripts/templates/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Template {
```
Comment (thanhnguyen-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/templates/index.js:1]: more specific naming

File: javascript-training/big-practice/pomodoro-app/src/scripts/templates/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  constructor() {}
+
+  static renderTask = task =>
+    `
+    <li class="task-item pointer" >
```
Comment (thanhnguyen-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/templates/index.js:6]: redundant space

File: javascript-training/big-practice/pomodoro-app/src/scripts/views/formView.js [hunk 3]
```diff
@@ -5,8 +5,13 @@ class FormView {
     this.taskForm = document.querySelector('.form-add-task');
     this.btnAddTask = document.querySelector('.btn-add-task');
     this.btnCancel = document.querySelector('.btn-cancel');
+    this.taskInput = document.querySelector('.task-input');
+    this.btnIncrease = document.querySelector('.btn-increase');
+    this.btnDecrease = document.querySelector('.btn-decrease');
+    this.inputElement = document.querySelector('.quantity-input');
```
Comment (thanhnguyen-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/views/formView.js:11]: more specific naming

File: javascript-training/big-practice/pomodoro-app/src/scripts/models/index.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
-import Task from './task';
+import TaskModel from './task';
 
 class Model {
```
Comment (tungtran-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/models/index.js:3]: Choice a better name

File: javascript-training/big-practice/pomodoro-app/src/scripts/models/index.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
   constructor() {
-    this.task = new Task();
+    this.taskModel = new TaskModel();
```
Comment (tungtran-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/models/index.js:5]: Task is enough

File: javascript-training/big-practice/pomodoro-app/src/scripts/controllers/index.js [hunk 6]
```diff
@@ -4,10 +4,20 @@ class Controller {
...
   initFormView() {
     this.view.formView.toggleTaskForm();
+    this.view.formView.submitAddTask(this.handleSubmitAddTask);
+    this.view.formView.controlValue();
   }
 
+  // Handle task addition by users.
+  handleSubmitAddTask = taskData => {
+    this.model.taskModel.addTask(taskData);
+    this.view.taskView.renderTask(this.model.taskModel.tasks);
```
Comment (thanhnguyen-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/controllers/index.js:17]: Put taskModel to controller constructure for shorten, then use as `this.taskModel`
```
this.taskModel = this.model.taskModel
```

**Review by thanhnguyen-agilityio — APPROVED** [review]

**Review by tungtran-agilityio — APPROVED** [review]

File: javascript-training/big-practice/pomodoro-app/src/scripts/views/taskView.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  constructor() {
+    this.taskList = document.querySelector('.task-list');
+  }
+
+  /**
+   *Display tasks using a template from Template.renderTask
+   * @param {Object} tasks
+   */
+  renderTask(tasks) {
+    this.taskList.innerHTML = '';
```
Comment (hungdo-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/views/taskView.js:13]: Can I ask why you do this

File: javascript-training/big-practice/pomodoro-app/src/scripts/templates/index.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+      <div class="task-wrapper">
+        <div class="task-item-done">
+          <span class="task-actual">0</span>
+          <span>/</span>
+          <span class="task-est">${task.inputValue}</span>
+        </div>
+
+        <button class="btn-more border-platinum pointer opacity">
+          <img src="${require('../../assets/icons/icon-edit.png')}" alt="icon more" class="icon-more edit">
```
Comment (hungdo-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/templates/index.js:21]: Please help import top of the file

File: javascript-training/big-practice/pomodoro-app/src/index.html [hunk 9]
```diff
@@ -35,44 +35,6 @@ <h1 class="title capitalize">tasks</h1>
 
       <ul class="task-list">
         <!-- TODO: This section will be render by template -->
```
Comment (hungdo-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/index.html:37]: Remove this TODO if you complete

File: javascript-training/big-practice/pomodoro-app/src/scripts/models/task.js [hunk 10]
```diff
@@ -1,8 +1,16 @@
...
+class TaskModel {
   constructor() {
-    // Initialize an empty array to store tasks.
-    this.task = [];
+    this.tasks = []; // Initialize an empty array to store tasks.
+  }
+
+  /**
+   *Add a task to the tasks array.
+   * @param {Object} taskData
```
Comment (hungdo-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/models/task.js:8]: What in this object can you help to specific more

File: javascript-training/big-practice/pomodoro-app/src/scripts/views/formView.js [hunk 11]
```diff
@@ -16,6 +21,43 @@ class FormView {
...
+  /**
+   * Handle the event when submitting the add task form
+   * @param {Object} handleSubmitAddTask
+   */
+  submitAddTask(handleSubmitAddTask) {
+    this.taskForm.addEventListener('submit', e => {
+      e.preventDefault();
+      const taskText = this.taskInput.value.trim();
+      const inputValue = this.inputElement.value.trim();
+      if (taskText && inputValue) {
```
Comment (hungdo-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/views/formView.js:39]: Can you help adding break line before each if condition, that more readable

File: javascript-training/big-practice/pomodoro-app/src/scripts/app.js [hunk 12]
```diff
@@ -4,7 +4,9 @@ import Controller from './controllers/index';
 
 export class App {
   start() {
-    const controller = new Controller(new Model(), new View());
+    const model = new Model();
```
Comment (thanhnguyen-agilityio) [inline javascript-training/big-practice/pomodoro-app/src/scripts/app.js:7]: One line as before is enough here :v

File: javascript-training/big-practice/pomodoro-app/src/scripts/views/taskView.js (hunk 7 above)
Comment (TIEN1909) [reply javascript-training/big-practice/pomodoro-app/src/scripts/views/taskView.js:13]: 'this' refers to an object View, then 'this.taskList' will point to the 'taskList' property of the View object.

Comment (hungdo-agilityio) [reply javascript-training/big-practice/pomodoro-app/src/scripts/views/taskView.js:13]: No I mean why you need to reassign value as an empty string :)

**Pull request merged**